### ISSUE TITLE
SceneView can freeze when invoking sceneView.startVideoRecording()

### ISSUE THREAD

**Issue by miketabb33** [gorastudio/SCNRecorder#22]
Just want to start by saying I think this library is really great, especially when compared to the few other sceneView recorders libraries. This library produces such a light weight and smooth recording experience. 

But,
Every so often, the scene view will freeze when invoking the sceneView.startVideoRecording() function, causing other areas of the UI to become unresponsive resulting in a force quit by the user. I tested on iPhone 8+, 4 out 25 AR record sessions experienced this freezing issue.

Is this a know issue? Has anybody else experienced library behavior as described above?

**Comment by v-grigoriev** [comment]
Hi @miketabb33, two questions for now.
Are you using the newest version i.e. 2.4.0? 
and do you call `sceneView.prepareForRecording()` at `viewDidLoad()`?

I faced this issue when AWAssetWriter blocked the main thread for a long time on a first run. I added a workaround for this a few versions ago.

Otherwise, it would be great if you run instruments, profile your app and find what is so time-consuming :)

**Comment by miketabb33** [comment]
Hi @v-grigoriev,
The current version specified in the pod file is ``` pod 'SCNRecorder', '~> 2.4' ```
Confirmed, the ```sceneView.prepareForRecording()``` is called at ``` viewDidLoad() ```

I profiled the app on a iPhone 7 and an iPhone SE (2g) and observed some leaks in CoreVideo. I A/B tested with another sceneView recording library and didn't find any leaks which makes me think the leaks are related to the library. Im also running instruments on a 2013 macBook pro, Im not sure if that could have anything to do with it lol. Also, not sure how related this could be in regards to my original issue report. I just want to share what I see :) I added some screen shots below for additional context. The screenshot with the completed responsible library and frame are from the iPhone 7.

Not sure how to how to read process times but my app is on the heavier side which may be a factor in freeze up. :)

<img width="1230" alt="Screen Shot 2021-02-05 at 10 03 06 AM" src="https://user-images.githubusercontent.com/36526318/107055238-1255b580-679f-11eb-9149-4ff4cf124dd9.png">
<img width="1230" alt="Screen Shot 2021-02-05 at 10 03 06 AM" src="https://user-images.githubusercontent.com/36526318/107055252-15e93c80-679f-11eb-87f1-78b3ff870d3e.png">
